### PR TITLE
[RLlib] DQN PolicyV2 sub-classing.

### PR DESCRIPTION
**Pull request by sven1977**
DQN PolicyV2 sub-classing for tf and torch policies.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/dqn/tests/test_dqn.py [hunk 1]
```diff
@@ -15,7 +16,7 @@
 class TestDQN(unittest.TestCase):
     @classmethod
     def setUpClass(cls) -> None:
-        ray.init()
+        ray.init(local_mode=True)  # TODO
```
Comment (charlesjsun) [inline rllib/algorithms/dqn/tests/test_dqn.py:19]: Debug statements left in

File: rllib/algorithms/dqn/tests/test_dqn.py [hunk 2]
```diff
@@ -26,7 +27,9 @@ def test_dqn_compilation(self):
         num_iterations = 1
         config = dqn.dqn.DQNConfig().rollouts(num_rollout_workers=2)
 
-        for _ in framework_iterator(config, with_eager_tracing=True):
+        for _ in framework_iterator(
+            config, frameworks="torch"
+        ):  # with_eager_tracing=True):
```
Comment (charlesjsun) [inline rllib/algorithms/dqn/tests/test_dqn.py:32]: Should revert to original to test all frameworks

File: rllib/algorithms/dqn/tests/test_dqn.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/dqn/tests/test_dqn.py:19]: thanks for the catch! :)

Comment (sven1977) [reply rllib/algorithms/dqn/tests/test_dqn.py:19]: done

File: rllib/algorithms/dqn/tests/test_dqn.py (hunk 2 above)
Comment (sven1977) [reply rllib/algorithms/dqn/tests/test_dqn.py]: great catch!
done

**Review by gjoliver — APPROVED** [review]

File: rllib/algorithms/dqn/dqn_tf_policy.py [hunk 3]
```diff
@@ -1,486 +1,372 @@
...
+    """Construct a SimpleQTFPolicy inheriting either dynamic or eager base policies.
 
-class ComputeTDErrorMixin:
-    """Assign the `compute_td_error` method to the DQNTFPolicy
+    Args:
+        base: Base class for this policy. DynamicTFPolicyV2 or EagerTFPolicyV2.
 
-    This allows us to prioritize on the worker side.
+    Returns:
+        A TF Policy to be used with MAMLTrainer.
```
Comment (charlesjsun) [inline rllib/algorithms/dqn/dqn_tf_policy.py:42]: "with DQN" instead of MAMLTrainer

**Pull request closed without merging**